Нужно больше функций

jiykka opened the pull request:
---
:mortar_board: [Нужно больше функций](https://up.htmlacademy.ru/javascript-individual/1/user/2054473/tasks/6)

:boom: https://htmlacademy-javascript.github.io/2054473-kekstagram-1/2/

jiykka committed:
Первые функции

saimon395 commented on index.html:
```diff
@@ -16,7 +17,8 @@
       <h2 class="img-filters__title  visually-hidden">Фильтр фотографий</h2>
       <form class="img-filters__form" action="index.html" method="get" autocomplete="off">
-        <button type=button class="img-filters__button  img-filters__button--active" id="filter-default">По умолчанию</button>
+        <button type=button class="img-filters__button  img-filters__button--active" id="filter-default">По
+          умолчанию</button>
```
а зачем разметку менял ?

saimon395 commented on js/function.js:
```diff
@@ -0,0 +1,32 @@
+  sameWord = sameWord.replaceAll(' ', '').toLowerCase();
+      const LAST_INDEX = sameWord.length - 1;
+      for (let i = 0; i < sameWord.length / 2; i++) {
+           //console.log(sameWord[i]);
```
закоментированный код лучше не оставлять, особенно консоль логи, как только отладку закончил , удаляй

saimon395 commented on js/function.js:
```diff
@@ -0,0 +1,32 @@
+
+   function stringTransform (sameString) {
+    sameString = sameString.replace(/[^0-9]+/g, '') || Math.abs(sameString);
+  return Number(sameString);
```
я бы какое нибудь более говорящее название функции дал, что то типа getNumberFromString

saimon395 commented on js/function.js:
```diff
@@ -0,0 +1,32 @@
+// Функция проверки на паллиндром
+
+function palindromeCheck(sameWord) {
```
я бы параметр просто string назвал

saimon395 commented on js/function.js:
```diff
@@ -0,0 +1,32 @@
+      return true;
+  }
+  palindromeCheck('топот твовт топоТ');
+
+
+  // Функция, которая принимает строку, извлекает содержащиеся в ней цифры от 0 до 9 и возвращает их в виде целого положительного числа.
```
не хватает ещё одной функции которая в задании требовалась, последняя из общего задания для всех проектов, та что функцию добавочными символами до нужной длины доводит

jiykka committed:
Исправления и третья функция

jiykka committed:
upd...

На счет изменения index.html - у меня редактор автоматом при сохранении расставляет пробелы и длину строк. С пред курсов остались эти настройки.

saimon395 commented on js/function.js:
```diff
@@ -0,0 +1,41 @@
+getNumberFromString('tgerg ef -4543.6,4');
+
+// Функция принимающая три параметра и возвращающая исходную строку, дополненную указанными символами до заданной длины
+const MAKE_NEW_STRING = (string, maxLength, symbol) => {
+  const DIFFERENCE = maxLength - string.length;
+  let newString = ''.concat(string);
+  let newSymbol = symbol;
+  if (string.length >= maxLength) {
```
по алгоритму вопросов нет, но есть по неймингу ), не стоит функции капсом называть, вообще капсом называют те константы которые до начала выполнения программы изместны, типа там число пи, или например ссылку на сервер, все остальные константы, константы лишь потому что мы значения не меняем, а не потому что это что то не изменяемое по сути, такие вещи называем как и обычные переменные кэмел кейсом

saimon395 replied:
с CHECK_LENGTH тоже самое

jiykka committed:
upd const naming
